### DISCUSSION TITLE
the difference between edge_index and edge_label_index

### DISCUSSION THREAD

**Discussion by knitemblazor** [pyg-team/pytorch_geometric#5189] [Q&A] [answered]
Hi i am trying to implement a link prediction model and using 

https://github.com/pyg-team/pytorch_geometric/blob/master/examples/link_pred.py

but i am unable to understand the difference between edge_index and edge_label index

**Comment by saiden89** [comment, marked as answer]
Hi, `edge_index` is the default name for the *message passing* edges index, that is, the set of edges we allow the model to send messages across. `edge_label_index` refers to the so called *supervision* edges: the ground truth values we want to compare our model's performance against, with consecutive loss calculation and backpropagation. You can read more [here](https://zqfang.github.io/2021-08-12-graph-linkpredict/).

**Reply by knitemblazor** [reply]
Thank you very much for the help. just one more doubt so if these edges used for message passing is absent will the gcn still be able to make link predictions and can multiple graphs be trained using the same approach instead of a single graph.
@saiden89

**Reply by saiden89** [reply]
Well if there are no message passing edges then we are not really dealing with a *Graph* Neural Network. One of the core ideas of GNN is that connected nodes pass messages according to network topology. If there is no index...then there is not really a network, or am I reading your question wrong?
For the second part, depends on what you mean by "multiple graphs". If your idea is having different kinds of nodes/edges, then you should take a look [here](https://pytorch-geometric.readthedocs.io/en/latest/notes/heterogeneous.html).

**Reply by knitemblazor** [reply]
Ok now i have a better understanding of the same i will have to formulate the problem in a different manner thank you for the response. I thought that weights of the edges that is connections are learned on the go with the given node representations .
Didnt know that these explicit message passing links were required for the same.
